crates/matrix-sdk/src/client/builder.rs

```diff
@@ -417,6 +417,16 @@ pub enum ClientBuildError {
     /// Error doing an HTTP request.
     #[error(transparent)]
     Http(#[from] HttpError),
+
+    /// Error opening the indexeddb store.
```

Comment:
Didn't we want to get away from sled/indexeddb store specific errors? This is also unused, I guess users might find this useful if they are opening the store while they are building the client, but what about custom store implementations?

Reply:
For me, as long as we re-export the stores from the `matrix-sdk` crate, it makes sense to provide a nice way to handle those errors.

If we decide to not export the stores anymore from the main crate and keep them as separate crates, of course this should go away too.

A custom store implementation should have custom errors anyway so it'll have custom handling.

Reply:
I was thinking the same, but two things are different here now compared to what we had before that convinced me, this is okay:
1. this isn't in base anymore, but in the higher level client
2. which is made for convenience and now decides upon the default, but doesn't have a proper error to go with.

As a result, If you were to set up the otherwise supported store just slightly differently, you'd have to wrap the error... that sounds unnecessarily inconvenient. On the other hand, if you were to use a not fully supported store, you'd have to manage the set up and its error management locally anyways and these aren't built in... so no one cares. 

While I agree (and made the point initially) that having wrapping errors for plugable crates always gives others some second-tier-usage parameter, the highest level crate is also very much about convenience for the user and I am okay with this (in the same way as just using features provides handy defaults that other implementations won't get).

Reply:
We might want to provide some nice example that can handle external stores as well (e.g. just wrapping in anyhow or eyre) and that'd be fine.